File: src/java/Mapping.java
public class Mapping{
	char[] key;
	
	public Mapping() {
		key = new char[26];
	}
	
	public Mapping(String cipher, String translation) {
		key = new char[26];

		for(int i = 0; i < translation.length(); i++)
		{
			key[(int)cipher.charAt(i)-(int)'A'] = translation.charAt(i);
		}
		//randomMapping();
	}
	
	public Mapping(Mapping m){
		key = new char[26];
		
		for(int i = 0; i < 26; i ++)
		{
			key[i] = m.getKey()[i];
		}
	}
	
	public char[] getKey() {
		return key;
	}
	
	public void setKey(char[] newKey) {
		key = newKey;
	}
	
	public void print() {
		for(int i = 0; i < 26; i ++)
		{
			System.out.print(key[i]);
		}
		System.out.println();
	}
}

File: src/py/Mapping.py
#!/usr/bin/env python
""" generated source for module Mapping """
class Mapping(object):
    """ generated source for class Mapping """
    key = []

    def __init__(self):
        """ generated source for method __init__ """
        self.key = ["#"]*26

    @classmethod
    def fromTranslation(self, cipher, translation):
        """ generated source for method __init___0 """
        newKey = ["#"]*26
        for i in range(len(translation)):
            newKey[ord(cipher[i]) - ord('A')] = translation[i]
        return newKey[:]

    def getKey(self):
        """ generated source for method getKey """
        return self.key

    def setKey(self, newKey):
        """ generated source for method setKey """
        self.key = newKey

